### ISSUE TITLE
Sound volume too low

### ISSUE THREAD

**Issue by tiagombrito** [SmartHome-yourself/sonoff-tx-ultimate-for-esphome#13]
Hi,

I noticed you managed to have sound playing in the Sonoff, but the sound is too low volume, even with volume at max settings in home assistant.

Sonoff's promo videos mention alarm siren as a use case, which was my need, is there anything that can be done to increase the sound volume further?

Thank you.

**Comment by danielscheidler** [comment]
I don't think that it can work as a real siren. 
At least I don't know of anything to get the audio output significantly louder. I'm happy to test with an unflashed one, but I suspect that these will not be louder either.

**Comment by tiagombrito** [comment]
Hi Daniel,

[This](https://www.youtube.com/watch?v=AGoKMUNvN-w&t=35s) is the youtube video I was refering to from Sonoff's channel, I just hope they didn't remove this from the specs, it'd be a cool feature to have.

Thank you very much for your reply.

**Comment by tiagombrito** [comment]
I bought some more wall switches, and i can't find any option in ewelink to make volume higher. 
So maybe this is a feature they discarded, and that video is not correct anymore.

**Comment by loadedlouie** [comment]
Hi , i tryed the sound from the switch, but although it works, it adds some "rain" to the sound itself, a bit, white noise if you can say that.. Also using for example google TTS it also adds that white noise sound ... is there any way to fix this ?

**Comment by loadedlouie** [comment]
> Hi , i tryed the sound from the switch, but although it works, it adds some "rain" to the sound itself, a bit, white noise if you can say that.. Also using for example google TTS it also adds that white noise sound ... is there any way to fix this ? Also great job doing this, I also add some other light effects to it, and I love it... thanks in advance for your time dedication and effort to create this for esphome/home assistant.

In case anyone is wondering...  i2s_comm_fmt: lsb (found it on another thread BLE proxy) solves the rain but keep the sound low.. :D

**Comment by TeDeVPrime** [comment]
yeah the speaker is VERY low. with the official firmware the sound is excellent. you can play doorbell sounds and it is just fine. in this case through ESP, you can barely hear it

**Comment by danielscheidler** [comment]
> In case anyone is wondering... i2s_comm_fmt: lsb (found it on another thread BLE proxy) solves the rain but keep the sound low.. :D

Yes, the i2s_comm_fmt 
is already updated in my variants too (did I miss one?). But I've no idea how to get it louder.

**Comment by TeDeVPrime** [comment]
i am not even sure if it is louder in the tasmota. the sound is like 10% of the volume even though Home Assistant says 100%

**Comment by conrad784** [comment]
I today flashed esphome without ever using it with ewelink and I am very happy about this project here.
Initially I did not think about using the speakers, but it would be nice to have. So I hope what was said here is true that the original firmware does something differently.

> yeah the speaker is VERY low. with the official firmware the sound is excellent. you can play doorbell sounds and it is just fine. in this case through ESP, you can barely hear it

May I ask how you found out which of the GPIOs are doing what @danielscheidler ? Maybe there is some pin which only has to be pulled high/low to enable some amplification?

**Comment by barth11** [comment]
@letsautomatenet  - The speaker is working flawlessly. I can put there mp3 song you can understand words. It's not loud or very good but that's not something I'd expect from something so small. For ringtone, notifications, doorbell it's enough.

**Comment by Whatsek** [comment]
> @letsautomatenet - The speaker is working flawlessly. I can put there mp3 song you can understand words. It's not loud or very good but that's not something I'd expect from something so small. For ringtone, notifications, doorbell it's enough.

I want to use it as a doorbell, but the volume on my device is very low. If In play a mp3, I can nearly hear it when my ear is 20cm from the device. Could you share your config?

**Comment by letsautomatenet** [comment]
> @letsautomatenet - The speaker is working flawlessly. I can put there mp3 song you can understand words. It's not loud or very good but that's not something I'd expect from something so small. For ringtone, notifications, doorbell it's enough.

Is there anything special you did? The audio section is as follows, as per the file tx_ultimate_local.yaml:

```
media_player:
  - platform: i2s_audio
    id: media_out
    name: ${friendly_name} Player
    dac_type: external
    i2s_dout_pin: ${audio_sdata_pin}
    i2s_audio_id: audio_i2s
    i2s_comm_fmt: lsb    
    mode: mono

``` 

The documentation here also says it doesn't work at moment so would be good to know how you got it working!

https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome#media-player

Thanks.

**Comment by barth11** [comment]
> Is there anything special you did?

@letsautomatenet - I didn't change anything. I just used the setup from this repo and played mp3 from HA.

![image](https://github.com/SmartHome-yourself/sonoff-tx-ultimate-for-esphome/assets/40197317/6c8a3117-d096-4e83-9b27-6522a14b875a)

@Whatsek  - I can hear the sound from another room but it's not loud but for me it was enough as it's not my primary speaker for ringbell. I didn't change anything.

**Comment by letsautomatenet** [comment]
Thanks for the reply.

No idea why mine isn't working. :-( It just goes straight to "Idle" when I try TTS or MP3.

**Comment by letsautomatenet** [comment]
I finally got it to work! Here were my issues in case someone else has similar:

- The URL presented by Home Assistant is my https external URL, as this is what is required for TTS with Google Home. This was not resolving/being blocked by my firewall for my ESP device. I assigned the ESP device a DNS address of my local AdGuard DNS server with a local rewrite rule to the HA ip address to prevent any potential DNS loopback/hairpin issues

- Some of my MP3s simply don't play on it. So when I thought I hadn't resolved the comms issue, I in fact had. Perhaps it's the encoding of these MP3s it doesn't like. The same files play on the Google speakers.

It is indeed pretty quiet.

**Comment by Whatsek** [comment]
I will test some MP3`s tonight, @letsautomatenet pretty quit, can you hear it from 3 meters away? (I cant)

**Comment by knuppe** [comment]
Same problem here, I can't hear the device from 3 meters away.

I was thinking about exploring the original firmware and see if we find something for it, does anyone have its original `.bin` file?

**Comment by Nalith** [comment]
> Same problem here, I can't hear the device from 3 meters away.
> 
> I was thinking about exploring the original firmware and see if we find something for it, does anyone have its original `.bin` file?

I have the US version firmware for 2G and 3G.

**Comment by robbiedehand** [comment]
for me the sound is not hearable, if i put my ear against is i hear some static and when i play media that static is changing in tone for a short while, anybody got suggestions?

for now i connected out the audio component and will install, see if the static is gone

**Comment by the-mentor** [comment]
Anyone knows what's the i2c audio chip by any chance?

**Comment by the-mentor** [comment]
I found this from Jessie on the esphome discord
```
`media_player` volume is software driven, but if the amp itself has a default low max volume setting and can be configured via i2c etc then it needs its own compoennt to do that configuration at boot
```

Maybe we need to figure it out somewhere

**Comment by robbiedehand** [comment]
So i figured i did some testing while the wifey was away, with a glass of whiskey and no pressure what so ever :-)

my findings in my case, tts or media browser from home assistat was using https, and my home assistant is https , local also for som reason it would not play, and so i made an test scenario which server the files on my NAS on a simple http server

simple as this.

`alias: Nieuwe automatisering
description: ""
trigger:
  - type: occupied
    platform: device
    device_id: 19c7bf41f11e0c024dbde6588a6ad5b6
    entity_id: a824329a64949469b6f7c99fca669ce8
    domain: binary_sensor
condition: []
action:
  - service: media_player.play_media
    metadata: {}
    data:
      media_content_id: http://192.168.100.202/files/2.mp3
      media_content_type: music
    target:
      entity_id: media_player.tx_tom_tx_tom_player
mode: single
`

now for the fun part i hav multiple files there;
ring.mp3
ring2.mp3 same as ring.mp3 but increase the volume of the file via mp3loader.com
1.mp 3, which is a MONO mp3
2.mp3 which is the same as 1.mp 3 but again i increase the volume of the file via mp3loader.com 4 decibels
3.mp3 which is the same as 1.mp 3 but again i increase the volume of the file via mp3loader.com 10 decibels

**Comment by xHirscHx** [comment]
Anybody fig this out yet when using tts. I bought a bigger speaker and it works but would prefer the smaller original  one.

**Comment by the-mentor** [comment]
@xHirscHx what do you mean by you bought a bigger speaker ? 
did you replace the speaker in the device ?

**Comment by xHirscHx** [comment]
yes, much bigger,  i 3d printed an case , im not using the tough part  just voice and the rgbs for my project

**Comment by xHirscHx** [comment]
> @xHirscHx what do you mean by you bought a bigger speaker ? did you replace the speaker in the device ?

yes, much bigger, i 3d printed an case , im not using the tough part just voice and the rgbs for my project

**Comment by the-mentor** [comment]
@xHirscHx do you mind sharing how you connected the speaker to the switch ?
thanks

**Comment by xHirscHx** [comment]
Took it apart bought 1mm 2 pin conecters on Amazon.

**Comment by QapaMcFly** [comment]
@xHirscHx  Please let us know how it went 
@robbiedehand  Thank you thats decent solution in my case

**Comment by xHirscHx** [comment]
So i added a  Max98357 I2S 3W to my setup and all works great now, it wont fit in stock but im using my tx ultimate just for the relays ,rgbs and printed a cover for a sc01 screen running  openhasp fits and looks great

**Comment by robbiedehand** [comment]
Jus a callback i have purchased two more of these device with "stock" firmware, nowere there is an option to play sound in the ewlink app, however i cant even find the frimra update option ewlink is bashing me about to upgrade the firmware. in home assistant with the ewlink integration same thin, no sound.

what is interesting thoug is that in DIY mode (lan mode ewlink integration in home assistant) one of my ESPhome flashed devices show up, that is exactly the only one that is sometimes disconnected from the wifi, for some reason i think it is almost a dual boot it show in ewlink and has ephome on it, but sorry i diverge, that one shows (however not availeble) a alarm and alarm bell

![Image](https://github.com/user-attachments/assets/ea34da0c-706b-442b-86b8-9d7fde69ddbb)

which indicates for me at least that these are just files sitting in the device and it is nog possible to cast any sound files normally with the stock firmware.

i will try to sftp or somthing into that if i can get back on that

**Comment by FedePrado** [comment]
I followed @robbiedehand 's details and worked like a charm.

Just an advice for you all.

The ESPHome Device Builder version 2026.4.0 **breaks everything**.

I rolled back to 2026.3.3 while I find a solution.

**Comment by danielscheidler** [comment]
@FedePrado please try again with current version. 
I think 326b9e8 should fix 2026.4
